tests/xharness/TestTasks/RunDevice.cs

```diff
@@ -103,6 +106,7 @@ public async Task RunTestAsync ()
 						timeoutMultiplier: testTask.TimeoutMultiplier,
 						variation: testTask.Variation,
 						buildTask: testTask.BuildTask);
+					testTask.Runner.UseTcpTunnel = useTcpTunnel && testTask.Device.DevicePlatform == DevicePlatform.iOS || testTask.Device.DevicePlatform == DevicePlatform.tvOS;
```

Comment:
Looks like you're silently ignoring the requested use of the tunnel.
Would not it be better to warn/error earlier that the configuration does not make sense ?